main.py

```diff
@@ -181,7 +181,7 @@ def main():
 
         # Sort the best layer-1 layouts and only return the best ones
         bestLayouts_L1_L2, bestScores_L1_L2 = getTopScores(
-            tempLayoutList, tempScoresList)
+            tuple(tempLayoutList), tempScoresList)
```

Comment:
`getTopScores` already returns the layouts as a `tuple`. Due to this fact, this line seems redundant.

Reply:
The return value is correct (tuple of two values). It's about the first input argument: `getTopScores` expects a `tuple` as first input but `tempLayoutList` is initialized as a `list` in line 100.

Reply:
Meaning even though the function returns `tuple<tuple, array>`, the first value gets converted into a list by the receiver? That's janky!

Could you try explicitly declaring the tuple's inner types with something [like this](https://stackoverflow.com/questions/47533787/typehinting-tuples-in-python#47534998)? I know I failed trying this about a year ago, but you might succeed with your superior python skills.

If it doesn't work in that elegant way, I'm open to implementing the change you proposed.

Reply:
Maybe we talk past each other :-)

Here, the first input parameter (=`layout`) of `getTopScores` is defined as a `tuple`:

```py
def getTopScores(layouts: tuple, scores: array, nrOfBest=NR_OF_BEST_LAYOUTS) -> tuple:
```

And here, the first input argument (=`tempLayoutList`) of `getTopScore` is a `list` (as defined in line 100):

```py
bestLayouts_L1_L2, bestScores_L1_L2 = getTopScores(tempLayoutList, tempScoresList)
```

And Cython gave me an error because `tempLayoutList`/`layout` is expected to be a `tuple`.
This is why I'm converting `tempLayoutList` into a `tuple`.

Reply:
Ah, that makes perfect sense!